requirements.txt

```diff
@@ -1,7 +1,7 @@
-matplotlib==3.3.4
+matplotlib==3.7.0
 numpy==1.23.0
-pandas==1.1.5
-scikit-learn==0.24.2
+pandas==2.0.0
```

Comment:
Should we bump up to the [current version](https://pandas.pydata.org/docs/whatsnew/v2.1.1.html)?

Reply:
It wouldn't be compatible with Python 3.8.

Should we update that or leave this?

Reply:
I'm happy to require a more modern version of Python--let's do it.

Reply:
Updated to Pandas 2.1.2

Updated supported Python versions to 3.9-3.10. CircleCI checks pass for Python 3.9 and 3.10. Unit tests pass when run locally using Python 3.9 and in repo's supported Dockerfile with Python 3.10.

Reply:
Excellent! 

Any reason to exclude 3.11?

Reply:
I ran into some issues trying to run the code with Python 3.11.  More specifically, when you pip install the requirements file, you get the following error below. It appears as though when you install scikit-learn, it collects its' numpy dependency (numpy 1.23.2) which appears to be incompatible with Python 3.11. I tried this for both scikit-learn 1.3.1 and 1.3.2. We could continue to work through different versions of the packages, but I'm not sure it's a huge gain at this time.

#0 3.328 Collecting scikit-learn==1.3.2 (from -r requirements.txt (line 4))
#0 3.336   Downloading scikit-learn-1.3.2.tar.gz (7.5 MB)
#0 3.457      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.5/7.5 MB 63.2 MB/s eta 0:00:00
#0 4.227   Installing build dependencies: started
#0 23.44   Installing build dependencies: finished with status 'error'
#0 23.47   error: subprocess-exited-with-error
#0 23.47   
#0 23.47   × pip subprocess to install build dependencies did not run successfully.
#0 23.47   │ exit code: 1
#0 23.47   ╰─> [319 lines of output]
#0 23.47       Ignoring numpy: markers 'python_version == "3.10" and platform_system == "Windows" and platform_python_implementation != "PyPy"' don't match your environment
#0 23.47       Collecting setuptools
#0 23.47         Obtaining dependency information for setuptools from https://files.pythonhosted.org/packages/bb/26/7945080113158354380a12ce26873dd6c1ebd88d47f5bc24e2c5bb38c16a/setuptools-68.2.2-py3-none-any.whl.metadata
#0 23.47         Downloading setuptools-68.2.2-py3-none-any.whl.metadata (6.3 kB)
#0 23.47       Collecting wheel
#0 23.47         Obtaining dependency information for wheel from https://files.pythonhosted.org/packages/fa/7f/4c07234086edbce4a0a446209dc0cb08a19bb206a3ea53b2f56a403f983b/wheel-0.41.3-py3-none-any.whl.metadata
#0 23.47         Downloading wheel-0.41.3-py3-none-any.whl.metadata (2.2 kB)
#0 23.47       Collecting Cython<3.0,>=0.29.33
#0 23.47         Obtaining dependency information for Cython<3.0,>=0.29.33 from https://files.pythonhosted.org/packages/cf/a3/43bf5b3ba528a8fa2aa25dd48317e493b46ce838098605277b9efc7a0c1a/Cython-0.29.36-cp311-cp311-musllinux_1_1_x86_64.whl.metadata
#0 23.47         Downloading Cython-0.29.36-cp311-cp311-musllinux_1_1_x86_64.whl.metadata (3.1 kB)
#0 23.47       Collecting oldest-supported-numpy
#0 23.47         Obtaining dependency information for oldest-supported-numpy from https://files.pythonhosted.org/packages/f2/a0/e3ddf322257d6d8798b5f08b5160eb697e0cd7f61cb70ad674aabae18051/oldest_supported_numpy-2023.10.25-py3-none-any.whl.metadata
#0 23.47         Downloading oldest_supported_numpy-2023.10.25-py3-none-any.whl.metadata (9.8 kB)
#0 23.47       Collecting scipy>=1.5.0
#0 23.47         Obtaining dependency information for scipy>=1.5.0 from https://files.pythonhosted.org/packages/36/95/69e32b4691daa6a6fe625d2e3724a39dc9b910c7860e739b583798d3d127/scipy-1.11.3-cp311-cp311-musllinux_1_1_x86_64.whl.metadata
#0 23.47         Downloading scipy-1.11.3-cp311-cp311-musllinux_1_1_x86_64.whl.metadata (60 kB)
#0 23.47            ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 60.4/60.4 kB 5.4 MB/s eta 0:00:00
#0 23.47       Collecting numpy==1.23.2 (from oldest-supported-numpy)
#0 23.47         Downloading numpy-1.23.2.tar.gz (10.7 MB)
#0 23.47            ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.7/10.7 MB 63.0 MB/s eta 0:00:00
#0 23.47         Installing build dependencies: started
#0 23.47         Installing build dependencies: finished with status 'done'
#0 23.47         Getting requirements to build wheel: started
#0 23.47         Getting requirements to build wheel: finished with status 'done'
#0 23.47         Preparing metadata (pyproject.toml): started
#0 23.47         Preparing metadata (pyproject.toml): finished with status 'done'
#0 23.47       Downloading setuptools-68.2.2-py3-none-any.whl (807 kB)
#0 23.47          ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 807.9/807.9 kB 52.0 MB/s eta 0:00:00
#0 23.47       Downloading wheel-0.41.3-py3-none-any.whl (65 kB)
#0 23.47          ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 65.8/65.8 kB 10.7 MB/s eta 0:00:00
#0 23.47       Using cached Cython-0.29.36-cp311-cp311-musllinux_1_1_x86_64.whl (2.0 MB)
#0 23.47       Downloading oldest_supported_numpy-2023.10.25-py3-none-any.whl (4.9 kB)
#0 23.47       Downloading scipy-1.11.3-cp311-cp311-musllinux_1_1_x86_64.whl (36.6 MB)
#0 23.47          ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 36.6/36.6 MB 34.4 MB/s eta 0:00:00
#0 23.47       Building wheels for collected packages: numpy
#0 23.47         Building wheel for numpy (pyproject.toml): started
#0 23.47         Building wheel for numpy (pyproject.toml): finished with status 'error'
#0 23.47         error: subprocess-exited-with-error
#0 23.47       
#0 23.47         × Building wheel for numpy (pyproject.toml) did not run successfully.
#0 23.47         │ exit code: 1
#0 23.47         ╰─> [268 lines of output]
#0 23.47             setup.py:71: RuntimeWarning: NumPy 1.23.2 may not yet support Python 3.11.
#0 23.47               warnings.warn(
#0 23.47             Running from numpy source directory.
#0 23.47             setup.py:86: DeprecationWarning:
#0 23.47       
#0 23.47               `numpy.distutils` is deprecated since NumPy 1.23.0, as a result
#0 23.47               of the deprecation of `distutils` itself. It will be removed for
#0 23.47               Python >= 3.12. For older Python versions it will remain present.
#0 23.47               It is recommended to use `setuptools < 60.0` for those Python versions.
#0 23.47               For more details, see:
#0 23.47                 https://numpy.org/devdocs/reference/distutils_status_migration.html